### PR TITLE
[to #48] try to fix scheduler problem

### PR DESCRIPTION
**Pull request by zeminzhou**
Signed-off-by: zeminzhou <zhouzemin@ping.com>

### What problem does this PR solve?

Keyspans to add & keyspans to remove, must be overlapped. So to be correct, I think we must finish remove jobs first, and then dispatch the add jobs.[here](https://github.com/tikv/migration/pull/74#discussion_r837388845)

Issue Number: to #48

Problem Description: 

Keyspans to add & keyspans to remove, must be overlapped. So to be correct, I think we must finish remove jobs first, and then dispatch the add jobs

### What is changed and how does it work?
After each acquisition of the current keyspan, compare the current keyspan with the keyspan of the previous period to establish a corresponding relationship newKeySpan -> needRemoveKeySpan. The running and deployment of the newKeySpan task needs to wait for all the needRemoveKeySpan tasks to stop. This check will be done in Processor.

### Code changes

- Has exported function/method change
- Has exported variable/fields change
- Has persistent data change

### CODE REVIEW

**Comment by zeminzhou** [comment]
TODO: Unit Test.

**Comment by zeminzhou** [comment]
@pingyu @zz-jason PTAL~

File: cdc/cdc/owner/scheduler_v1.go [hunk 1]
```diff
@@ -113,6 +118,27 @@ func (s *oldScheduler) Tick(
 	return shouldUpdateState, nil
 }
 
+func (s *oldScheduler) diffCurrentKeySpans(currentKeySpans map[model.KeySpanID]regionspan.Span) (map[model.KeySpanID]struct{}, []model.KeySpanID) {
```
Comment (pingyu) [inline cdc/cdc/owner/scheduler_v1.go:121]: Please add unit test for this method.

File: cdc/cdc/owner/scheduler_v1.go [hunk 2]
```diff
@@ -246,26 +272,41 @@ func (s *oldScheduler) dispatchToTargetCaptures(pendingJobs []*schedulerJob) {
 
 // syncKeySpansWithCurrentKeySpans iterates all current keyspans to check whether it should be listened or not.
 // this function will return schedulerJob to make sure all keyspans will be listened.
-func (s *oldScheduler) syncKeySpansWithCurrentKeySpans() ([]*schedulerJob, error) {
+func (s *oldScheduler) syncKeySpansWithCurrentKeySpans(newKeySpans map[model.KeySpanID]struct{}, needRemoveKeySpans []model.KeySpanID) ([]*schedulerJob, error) {
```
Comment (pingyu) [inline cdc/cdc/owner/scheduler_v1.go:275]: Suggest to add unit test for this method.

File: cdc/cdc/processor/processor.go [hunk 3]
```diff
@@ -597,6 +601,18 @@ func (p *processor) handleKeySpanOperation(ctx cdcContext.Context) error {
 	return nil
 }
 
+func (p *processor) checkRelatedKeyspans(relatedKeySpans []model.KeySpanLocation) bool {
```
Comment (pingyu) [inline cdc/cdc/processor/processor.go:604]: Please add unit test for this method.

File: cdc/pkg/scheduler/workload.go [hunk 4]
```diff
@@ -88,7 +88,7 @@ func (w workloads) Skewness() float64 {
 }
 
 func (w workloads) SelectIdleCapture() model.CaptureID {
-	minWorkload := uint64(math.MaxUint64)
+	minWorkload := uint64(math.MaxInt64)
```
Comment (haojinming) [inline cdc/pkg/scheduler/workload.go:91]: maybe revert this change?

Comment (zeminzhou) [reply cdc/pkg/scheduler/workload.go]: Done

**Review by pingyu — APPROVED** [review]
LGTM~

**Review by haojinming — APPROVED** [review]
LGTM~
